openapi.yml

```diff
@@ -0,0 +1,475 @@
+          schema: {type: string}
+          required: true
+      requestBody:
+        description: 'Replace existing or create new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+    get:
+      tags: ['workflow', 'discovery']
+      operationId: getWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return ProcessorJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/Workflow'}}
+            application/vnd.ocrd+zip: {}
+        '404': {content: {}, description: 'Workflow not available'}
+    post:
+      tags: ['workflow']
+      operationId: runWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return WorkflowJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/WorkflowJob'}}
+
+  '/workflow/{workflow-id}/{job-id}':
+    get:
+      tags: ['workflow']
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      operationId: getWorkflowJob
+      responses:
+        '200':
+          description: Return WorkflowJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/WorkflowJob'}}
+        '404':
+          description: WorkflowJob not found
+          content: {}
+
+  '/workspace':
+    get:
+      tags: ['workspace']
+      operationId: getWorkspaces
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Workspace'
+    post:
+      tags: ['workspace']
+      summary: Post a new workspace
+      operationId: createWorkspace
+      requestBody:
+        description: OCRD-ZIP of the new workspace
+        content:
+          'application/vnd.ocrd+zip': {}
+        required: true
+      responses:
+        201:
+          description: Workspace created
+          content:
+            application/json: {schema: {$ref: '/components/schemas/Workspace'}}
+
+  '/workspace/{workspace-id}':
+    put:
+      tags: ['workspace']
+      summary: Replace an existing workspace
```

Comment:
In `/workflow/{job-id}` where we have a similar structure the summary says "Replace existing or create new workflow". Isn't this a similar case calling for a similar summary?

Reply:
Yes

Reply:
```suggestion
      summary: Replace existing or create new workspace
```

Reply:
> Replace existing or create new workspace

Creating a new workspace by putting to a specific ID might cause ID clashes. To post a new workspace, `POST` seems the right way and I see no need to duplicate the functionality. Or do you have a particular use case in mind?

Reply:
No. Then we leave it as is.